course-v2/layouts/partials/course_description.html

```diff
@@ -1,5 +1,8 @@
 {{ $courseData := .context.Site.Data.course }}
-{{ $shouldCollapseDescription := gt (len $courseData.course_description) 320 }}
+{{ $shouldCollapseDescription := false }}
+{{ with $courseData.course_description }}
+  {{ $shouldCollapseDescription := gt (len .) 320 }}
```

Comment:
If we go with this approach, when you reassign variables in a Go template you need to drop the `:`. So, it should look like this:

```
{{ $shouldCollapseDescription := false }}
{{ with $courseData.course_description }}
  {{ $shouldCollapseDescription = gt (len .) 320 }}
{{ end }}
```

Using `with` is a fine approach when you are not sure if the variable will exist, and this will ensure that `$shouldCollapseDescription` will default to false. The approach @MAbdurrehman12 suggested would work as well:

```
{{ $courseDescription := $courseData.course_description | default "" }}
{{ $shouldCollapseDescription = gt (len $courseDescription) 320 }}
```

In the scope of just addressing the course description I don't have an issue with either of these approaches, but I think that we should look at the big picture and come up with a solution for error handling when it comes to fetching any course level metadata that might be un-set like I described above.

Reply:
P.S. The larger issue of checking all of our course level metadata before accessing doesn't have to be completed in this PR btw, I'm fine with just merging this once the syntax is corrected here. In a later PR we should address the larger issue of sanely handling errors for any metadata fetching.